onnxruntime/core/providers/cuda/nn/pool.cc

```diff
@@ -164,7 +164,7 @@ Status Pool<T, PoolType>::ComputeInternal(OpKernelContext* context) const {
   }
 
   cudnnPoolingMode_t mode = CUDNN_POOLING_MAX;
-  if (PoolType::type == onnxruntime::PoolType::kAveragePool) {
+  if constexpr (PoolType::type == onnxruntime::PoolType::kAveragePool) {
```

Comment:
Why not ORT_IF_CONSTEXPR  ?

Reply:
It's a cc file. ORT_IF_CONSTEXPR  is for header files that are also used in *.cu files. This file is compiled by GCC which can support C++17 well. The main limitation comes from nvcc, which compiles *.cu files.

Reply:
But, we can use ORT_IF_CONSTEXPR  in all the places. Was that your question?

Reply:
Yes :)